wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt

```diff
@@ -0,0 +1,343 @@
+            |            null,
+            |    val unknownFields: ByteString = ByteString.EMPTY
+            |) : Message<Dinosaur, Dinosaur.Builder>(ADAPTER, unknownFields) {
+            """.trimMargin()
+        )
+  }
+
+  @Test
+  fun pruneRemovesField() {
+    val fixtureRoot = File("src/test/projects/field-prune")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+
+    val generatedProto =
+      File(fixtureRoot, "build/generated/src/main/java/com/squareup/dinosaurs/Dinosaur.kt")
+    assertThat(generatedProto).exists()
+
+    assertThat(generatedProto.readText())
+        .doesNotContain(
+            """
+            |    /**
+            |     * Common name of this dinosaur, like "Stegosaurus".
+            |     */
+            |    @field:WireField(tag = 1, adapter = "com.squareup.wire.ProtoAdapter#STRING") val name: String? =
+            |            null,
+            """.trimMargin()
+        )
+  }
+
+  @Test
+  fun ruleKeepsField() {
+    val fixtureRoot = File("src/test/projects/field-rule-root")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .doesNotContain("Writing com.squareup.geology.Period")
+
+    val generatedProto =
+      File(fixtureRoot, "build/generated/src/main/java/com/squareup/dinosaurs/Dinosaur.kt")
+    assertThat(generatedProto).exists()
+
+    assertThat(generatedProto.readText())
+        .contains(
+            """
+            |data class Dinosaur(
+            |    /**
+            |     * Common name of this dinosaur, like "Stegosaurus".
+            |     */
+            |    @field:WireField(tag = 1, adapter = "com.squareup.wire.ProtoAdapter#STRING") val name: String? =
+            |            null,
+            |    val unknownFields: ByteString = ByteString.EMPTY
+            |) : Message<Dinosaur, Dinosaur.Builder>(ADAPTER, unknownFields) {
+            """.trimMargin()
+        )
+  }
+
+  @Test
+  fun ruleRemovesField() {
+    val fixtureRoot = File("src/test/projects/field-rule-prune")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+
+    val generatedProto =
+      File(fixtureRoot, "build/generated/src/main/java/com/squareup/dinosaurs/Dinosaur.kt")
+    assertThat(generatedProto).exists()
+
+    assertThat(generatedProto.readText())
+        .doesNotContain(
```

Comment:
Same here. Generally, `contains()` tests will start failing if generated code changes - it's kinda fine, we'll need to fix them up. `doesNotContain()` tests though are not safe cause they'll turn green if generated code changes.

Reply:
given that these test fixtures are constant, the generated code will change when we change the generator itself, which is known at the moment.  I assume the generator will stabilize soon + this proto has a small surface area, so I'm less worried.

Reply:
Check for less? The absence of `val name` seems pretty safe

Reply:
Updated `doesNotContain` tests to only check for `val name`.
Updated `contains` tests to parse field names via regex and check for `val name`.